notebook/static/notebook/js/codecell.js

```diff
@@ -335,10 +335,10 @@ define([
         this.render();
         this.events.trigger('execute.CodeCell', {cell: this});
         var that = this;
-        this.events.on('finished_iopub.Kernel', function (evt, data) {
+        this.events.one('finished_iopub.Kernel', function (evt, data) {
```

Comment:
However, I just realized that if another cell's execution completes in between this cell's execution starting and completing, then this event listener will be disposed before it can handle execution completion...